CONTRIBUTING.md

```diff
@@ -7,19 +7,101 @@ We're glad you want to contribute markdownlint! This document will help answer c
+
+[Beginner issues](https://github.com/markdownlint/markdownlint/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22+sort%3Acomments-desc) - issues which should only require a few lines of code, and a test or two.
+[Help wanted issues](https://github.com/markdownlint/markdownlint/issues?q=is%3aissue+is%3aopen+label%3a%22help+wanted%22+sort%3Acomments-desc) - issues which should be a bit more involved than beginner issues.
+Both issue lists are sorted by total number of comments. While not perfect, number of comments is a reasonable proxy for impact a given change will have.
+
+### Local Development
+
+**Under Development (Oct 2019)**
+<!--
+* Please try not to mess with the Rakefile, version, or history. If you want to have your own version, or is otherwise necessary, that is fine, but please isolate to its own commit so I can cherry-pick around it.
+* [ ] Run `bundle exec rake default`. It executes all tests and Markdownlint for itself, and generates the documentation
+-->
+
+### Pull Request Requirements
+
+Markdownlint strives to ensure high quality for the project. In order to promote this, we require that all pull requests to meet these specifications:
+
+* **Tests:** To ensure high quality code and protect against future regressions, we require tests for all new/changed functionality in Markdownlint. Test positive and negative scenarios, try to break the new code now.
+* **Green CI Tests:** We use [Travis CI](https://travis-ci.org/markdownlint/markdownlint) to test all pull requests. We require these test runs to succeed on every pull request before being merged.
+
+Before submitting the Pull Request, make sure the following are checked:
+
+* Used the same coding conventions as the rest of the project
+* Wrote [good commit messages](https://chris.beams.io/posts/git-commit/)
+* Commit message starts with `[Fix #issue-number]` if there is a corresponding open GitHub issue
+* Feature branch is up-to-date with `master`, if not - rebase it
+* All [related commits squashed together](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html)
+* Added tests for all new/changed functionality, including tests for positive and negative scenarios
+* Added an entry to the [Changelog](https://github.com/markdownlint/markdownlint/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [Changelog entry format](https://github.com/markdownlint/markdownlint/blob/master/CONTRIBUTING.md#changelog-entry-format)
+* The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences
+
+See [this article](https://help.github.com/articles/about-pull-requests/) if you're not familiar with GitHub Pull Requests.
+
+### Changelog entry format
+
+Here are a few examples:
+
+```
+* [#248](https://github.com/markdownlint/markdownlint/pull/248): Ignore blank line after front matter, closes [#247](https://github.com/markdownlint/markdownlint/issues/247) ([@apjanke][])
```

Comment:
This isn't how they've been doing it... and seems a bit more onerous than what's there... I don't feel strongly though

Reply:
I kind of liked this, or at least linking to the PR/issue for a quick reference but we could probably do this with proper release sheets and queries.

Reply:
I think linking to the PR is useful.. the PR links to the issues, I'm either-way on that. I don't think it's worth putting the author, that's just more work and already in the git log.

Reply:
Sure, sometimes including the author is a nice bit of recognition, but I'm fine dropping the issue number and such.

Reply:
I think listing them is good... I do like saying `Closes #XXX #XXX #XXX` dunno they all need to be links if the PR is a link. but I don't feel super strongly.